### PR TITLE
added feature to automatically change port

### LINKED ISSUE #2857: [REF-2220] reflex run should automatically select new port if default is taken
Currently if you have one app running, and try to `reflex run` on another app, it asks you to change the port. Instead, we should automatically find the next free frontend and backend ports (3001, 8001) and run it on that.

<sub>[REF-2220](https://linear.app/reflex-dev/issue/REF-2220/reflex-run-should-automatically-select-new-port-if-default-is-taken)</sub>

### PR DESCRIPTION
**Pull request by wassafshahzad**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [ ] Does your submission pass the tests? 
- [ ] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Description
Added feature to automatically change port if currently one is used.

### Linked Issue
Closes #2857

### CODE REVIEW

**Comment by srikary12** [comment]
I've implemented something similar, but added just auto select functionality after 5 sec instead of by default. You can find the changes here https://github.com/srikary12/reflex-dev

**Comment by wassafshahzad** [comment]
> I've implemented something similar, but added just auto select functionality after 5 sec instead of by default. You can find the changes here https://github.com/srikary12/reflex-dev

Are you shadowing the repo ? if not you should first have the issue  assigned to your self before starting and then create a pull request. Other then that I have taken a look at your changes. You are using sockets to check if the port is connected or not, which is not needed since we already have a function called `is_process_on_port` which informs us if the port is already occupied.

I do like you idea of informing the user before auto selecting the ports but I believe it would be better to have a property in the config something like "auto_select_port: Bool". Since asking the user `change_or_terminate_port` is a bit redundant in most case as the user can just manually enter a port.
Still good effort.

File: reflex/utils/processes.py [hunk 1]
```diff
@@ -87,41 +87,20 @@ def kill_process_on_port(port):
...
-                f"The {_type} will run on port [bold underline]{new_port}[/bold underline]."
-            )
-            return new_port
-    else:
-        console.log("Exiting...")
-        raise typer.Exit()
+    new_port = str(int(port) + 1)
+    if is_process_on_port(new_port):
+        return change_port(new_port)
+    return new_port
```
Comment (masenf) [inline reflex/utils/processes.py:103]: i think we should print an info message here, similar to the one that was removed

```python
            console.info(
                f"The {_type} will run on port [bold underline]{new_port}[/bold underline]."
            )
```

**Review by masenf — CHANGES_REQUESTED** [review]
lgtm

File: reflex/utils/processes.py (hunk 1 above)
Comment (wassafshahzad) [reply reflex/utils/processes.py:107]: @masenf Done

**Review by picklelo — APPROVED** [review]
Works great for me, this is a sweet improvement!

**Pull request merged**